File: app/src/androidTest/java/de/westnordost/streetcomplete/data/changesets/OpenChangesetsDaoTest.java
package de.westnordost.streetcomplete.data.changesets;

import org.junit.Before;
import org.junit.Test;

import de.westnordost.streetcomplete.data.ApplicationDbTestCase;

import static org.junit.Assert.*;

public class OpenChangesetsDaoTest extends ApplicationDbTestCase
{
	private OpenChangesetsDao dao;

	private static final OpenChangesetKey Q = new OpenChangesetKey("Hurzipurz","test");
	private static final OpenChangesetKey P = new OpenChangesetKey("Brasliweks","test");

	@Before public void createDao()
	{
		dao = new OpenChangesetsDao(dbHelper, null);
	}

	@Test public void deleteNonExistent()
	{
		assertFalse(dao.delete(Q));
	}

	@Test public void createDelete()
	{
		dao.replace(Q,1);
		assertTrue(dao.delete(Q));
		assertNull(dao.get(Q));
	}

	@Test public void getNull()
	{
		assertNull(dao.get(Q));
	}

	@Test public void insertChangesetId()
	{
		dao.replace(Q, 12);
		OpenChangesetInfo info = dao.get(Q);
		assertEquals(12, (long) info.changesetId);
		assertEquals(Q.questType, info.key.questType);
		assertEquals(Q.source, info.key.source);
	}

	@Test public void replaceChangesetId()
	{
		dao.replace(Q, 12);
		dao.replace(Q, 6497);
		assertEquals(6497, (long) dao.get(Q).changesetId);
	}

	@Test public void getNone()
	{
		assertTrue(dao.getAll().isEmpty());
	}

	@Test public void insertTwo()
	{
		dao.replace(Q,1);
		dao.replace(P,2);
		assertEquals(2,dao.getAll().size());
	}
}

File: app/src/androidTest/java/de/westnordost/streetcomplete/data/osm/upload/changesets/OpenChangesetsDaoTest.kt
package de.westnordost.streetcomplete.data.osm.upload.changesets

import org.junit.Before
import org.junit.Test

import de.westnordost.streetcomplete.data.ApplicationDbTestCase

import org.junit.Assert.*

class OpenChangesetsDaoTest : ApplicationDbTestCase() {
    private lateinit var dao: OpenChangesetsDao

    private val Q = "Hurzipurz"
    private val P = "Brasliweks"
    private val SOURCE = "test"

    @Before fun createDao() {
        dao = OpenChangesetsDao(dbHelper, OpenChangesetMapping())
    }

    @Test fun deleteNonExistent() {
        assertFalse(dao.delete(Q, SOURCE))
    }

    @Test fun createDelete() {
        dao.put(OpenChangeset(Q, SOURCE, 1))
        assertTrue(dao.delete(Q, SOURCE))
        assertNull(dao.get(Q, SOURCE))
    }

    @Test fun getNull() {
        assertNull(dao.get(Q, SOURCE))
    }

    @Test fun insertChangesetId() {
        dao.put(OpenChangeset(Q, SOURCE, 12))
        val info = dao.get(Q, SOURCE)!!
        assertEquals(12, info.changesetId)
        assertEquals(Q, info.questType)
        assertEquals(SOURCE, info.source)
    }

    @Test fun replaceChangesetId() {
        dao.put(OpenChangeset(Q, SOURCE, 12))
        dao.put(OpenChangeset(Q, SOURCE, 6497))
        assertEquals(6497, dao.get(Q, SOURCE)!!.changesetId)
    }

    @Test fun getNone() {
        assertTrue(dao.getAll().isEmpty())
    }

    @Test fun insertTwo() {
        dao.put(OpenChangeset(Q, SOURCE, 1))
        dao.put(OpenChangeset(P, SOURCE, 2))
        assertEquals(2, dao.getAll().size)
    }
}
